### PR TITLE
Initial port to nautobot 2

### PR DESCRIPTION
**Pull request by TobiPeterG**
Depends on https://github.com/nautobot/go-nautobot/pull/30

### CODE REVIEW

**Comment by TobiPeterG** [comment]
This is an initial bring up, not tested. As https://github.com/nautobot/go-nautobot/pull/30 is not merged yet, this code points to my go-nautobot repo.

How can I easily test the code?

**Comment by chadell** [comment]
> This is an initial bring up, not tested. As [nautobot/go-nautobot#30](https://github.com/nautobot/go-nautobot/pull/30) is not merged yet, this code points to my go-nautobot repo.
> 
> How can I easily test the code?

test it against your branch, and when it's ready, we update it here

**Comment by TobiPeterG** [comment]
> > This is an initial bring up, not tested. As [nautobot/go-nautobot#30](https://github.com/nautobot/go-nautobot/pull/30) is not merged yet, this code points to my go-nautobot repo.
> > 
> > How can I easily test the code?
> 
> test it against your branch, and when it's ready, we update it here

I tested it already with the demo Nautobot instance and everything seems to work. :)
I already added the functionality to add/read/update/delete cluster types, clusters and virtual machines

**Comment by TobiPeterG** [comment]
@chadell please have a look, update to use official go-nautobot :)

File: docs/data-sources/available_ip_address.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
```
Comment (chadell) [inline docs/data-sources/available_ip_address.md:2]: @TobiPeterG could you update the readme to explain how you have generated the docs. thanks

**Review by chadell — APPROVED** [review]
LGTM

File: docs/data-sources/available_ip_address.md (hunk 1 above)
Comment (TobiPeterG) [reply docs/data-sources/available_ip_address.md:2]: This is generated with the command "go generate" if I'm not mistaken, it's documented in the main.go

It is also automatically executed in the GitHub ci afaik :)

**Pull request merged**